activerecord/lib/active_record/relation.rb

```diff
@@ -657,7 +657,9 @@ def has_join_values?
       end
 
       def exec_queries(&block)
-        @records = eager_loading? ? find_with_associations.freeze : @klass.find_by_sql(arel, bound_attributes, &block).freeze
+        @records = uncached_if_necessary do
+          eager_loading? ? find_with_associations.freeze : @klass.find_by_sql(arel, bound_attributes, &block).freeze
+        end
 
         preload = preload_values
         preload += includes_values unless eager_loading?
```

Comment:
I think the `preload` call just down ⬇️ there wants to be wrapped too. Don't want to fill the cache with `SELECT * FROM children WHERE parent_id IN (long, list, of, ids)` for each batch.